### ISSUE TITLE
Adding interaction term with formula not compatible with skLearn.GridSearchCV 

### ISSUE THREAD

**Issue by SallyMostafa** [CamDavidsonPilon/lifelines#1164]
I'm using sklearn adapter to integrate the CPH model with my machine learning workflow (GridSearchCV, etc.). It works fine, however, when I add interactions "formula", it just ignored! 

I'm able to use the adapter with formula if I'm just fitting directly, but if I tried to use GridSearchCV it is just ignored. I sent it as a parameter to the model or as a param to the GridSearchCV and both did not work. 

Basically, I want to do this GridSearchCV lifelines example but with interaction: https://lifelines.readthedocs.io/en/latest/Compatibility%20with%20scikit-learn.html

**Comment by kyleabeauchamp** [comment]
I think I've found a related issue.  It seems that the only way to specify a formula is to call the .fit() function.  However, this seems to be incompatible with model comparison using the `utils.k_fold_cross_validation` function---that function seems to assume that each model object is self-contained.  One way around this might be to pass the formula into the object __init__ function.  A possibly API-compatible (but less elegant) way might be to allow the fitter_kwargs to be a list of dictionaries.

**Comment by dbikiel** [comment]
It will be very useful. The beauty of the formula approach is the automatic conversion of categoricals and name management of the dataframe.
